# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/txmgr/tests/TestTXMgr.cpp
# Commit: b9ebe58001d7
# Full Hash: b9ebe58001d787bb0bbad6d39ae8681966d5a77f
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-04-04 09:29:13
# Regressor Bug: 1619914
# File Overlap Count: 1
# Description:
#   Bug 1619914 - part 2: Mark transaction class methods and their caller methods as `MOZ_CAN_RUN_SCRIPT r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D69303
# ==============================================================================

diff -r ce2087ddab5c -r b9ebe58001d7 editor/txmgr/tests/TestTXMgr.cpp
--- a/editor/txmgr/tests/TestTXMgr.cpp	Fri Apr 03 08:30:37 2020 +0000
+++ b/editor/txmgr/tests/TestTXMgr.cpp	Fri Apr 03 08:32:01 2020 +0000
@@ -270,7 +270,7 @@
 
   ~SimpleTransaction() override = default;
 
-  NS_IMETHOD DoTransaction() override {
+  MOZ_CAN_RUN_SCRIPT NS_IMETHOD DoTransaction() override {
     //
     // Make sure DoTransaction() is called in the order we expect!
     // Notice that we don't check to see if we go past the end of the array.
@@ -286,7 +286,7 @@
     return (mFlags & THROWS_DO_ERROR_FLAG) ? NS_ERROR_FAILURE : NS_OK;
   }
 
-  NS_IMETHOD UndoTransaction() override {
+  MOZ_CAN_RUN_SCRIPT NS_IMETHOD UndoTransaction() override {
     //
     // Make sure UndoTransaction() is called in the order we expect!
     // Notice that we don't check to see if we go past the end of the array.
@@ -302,7 +302,7 @@
     return (mFlags & THROWS_UNDO_ERROR_FLAG) ? NS_ERROR_FAILURE : NS_OK;
   }
 
-  NS_IMETHOD RedoTransaction() override {
+  MOZ_CAN_RUN_SCRIPT NS_IMETHOD RedoTransaction() override {
     //
     // Make sure RedoTransaction() is called in the order we expect!
     // Notice that we don't check to see if we go past the end of the array.
@@ -372,7 +372,7 @@
 
   ~AggregateTransaction() override = default;
 
-  NS_IMETHOD DoTransaction() override {
+  MOZ_CAN_RUN_SCRIPT NS_IMETHOD DoTransaction() override {
     if (mLevel >= mMaxLevel) {
       // Only leaf nodes can throw errors!
       mFlags |= mErrorFlags;
@@ -390,7 +390,7 @@
     }
 
     if (mFlags & BATCH_FLAG) {
-      rv = mTXMgr->BeginBatch(nullptr);
+      rv = MOZ_KnownLive(mTXMgr)->BeginBatch(nullptr);
       if (NS_FAILED(rv)) {
         return rv;
       }
@@ -416,7 +416,7 @@
       RefPtr<AggregateTransaction> tximpl = new AggregateTransaction(
           mTXMgr, cLevel, i, mMaxLevel, mNumChildrenPerNode, flags);
 
-      rv = mTXMgr->DoTransaction(tximpl);
+      rv = MOZ_KnownLive(mTXMgr)->DoTransaction(tximpl);
       if (NS_FAILED(rv)) {
         if (mFlags & BATCH_FLAG) {
           mTXMgr->EndBatch(false);
@@ -482,7 +482,7 @@
 /**
  * Test behaviors in non-batch mode.
  **/
-void quick_test(TestTransactionFactory* factory) {
+MOZ_CAN_RUN_SCRIPT_BOUNDARY void quick_test(TestTransactionFactory* factory) {
   /*******************************************************************
    *
    * Create a transaction manager implementation:
@@ -497,7 +497,7 @@
    *
    *******************************************************************/
 
-  nsresult rv = mgr->DoTransaction(0);
+  nsresult rv = mgr->DoTransaction(nullptr);
   EXPECT_EQ(rv, NS_ERROR_NULL_POINTER);
 
   /*******************************************************************
@@ -1229,7 +1229,8 @@
 /**
  * Test behaviors in batch mode.
  **/
-void quick_batch_test(TestTransactionFactory* factory) {
+MOZ_CAN_RUN_SCRIPT_BOUNDARY void quick_batch_test(
+    TestTransactionFactory* factory) {
   /*******************************************************************
    *
    * Create a transaction manager implementation:
@@ -1858,7 +1859,8 @@
  * Create 'iterations * (iterations + 1) / 2' transactions;
  * do/undo/redo/undo them.
  **/
-void stress_test(TestTransactionFactory* factory, int32_t iterations) {
+MOZ_CAN_RUN_SCRIPT_BOUNDARY void stress_test(TestTransactionFactory* factory,
+                                             int32_t iterations) {
   /*******************************************************************
    *
    * Create a transaction manager:
